lib/loader.js

```diff
@@ -16,10 +16,14 @@ module.exports = function(content, map) {
 	var root = query.root;
 	var moduleMode = query.modules || query.module;
 	var camelCaseKeys = query.camelCase || query.camelcase;
+	var sourceMap = query.sourceMap || false;
 	var resolve = createResolver(query.alias);
 
-	if(map !== null && typeof map !== "string") {
+	if(sourceMap && map !== null && typeof map !== "string") {
 		map = JSON.stringify(map);
+	} else {
+        // Some loaders (example `"postcss-loader": "1.x.x"`) always generates source map, we should remove it
+		map = null;
```

Comment:
yep 😉 I know, but we need to make clear how we want to handle sourcemaps in the 'overall' picture and then fix it in the loaders e.g

- `{ sourceMap: false }` => no sourcemaps (absolutely, if one loader in the chain is `false`, discard them enterily)
- `{ sourceMap: true }` => as `map` (external)
- `{ sourceMap: 'inline' }` => in the CSS (`{String}`), not in favour of personally, but maybe some folks e.g Angular need it :stuck_out_tongue: